polkadot/node/core/backing/src/lib.rs

```diff
@@ -383,6 +383,19 @@ struct TableContext {
 	validator: Option<Validator>,
 	groups: HashMap<ParaId, Vec<ValidatorIndex>>,
 	validators: Vec<ValidatorId>,
+	disabled_validators: Vec<ValidatorIndex>,
+}
+
+impl TableContext {
+	pub fn validator_is_disabled(&self, validator_idx: &ValidatorIndex) -> bool {
+		self.disabled_validators
+			.iter()
+			.any(|disabled_val_idx| *disabled_val_idx == *validator_idx)
+	}
+
+	pub fn local_validator_is_disabled(&self) -> Option<bool> {
```

Comment:
Why do we need to return Option ?

Reply:
`self.validator` is an `Option`. Afair the code can be executed in cases where the local node is not a validator?

I decided to return an `Option` here too and handle it at the caller.